beacon/goclient/proposer.go

```diff
@@ -123,6 +163,86 @@ func (gc *goClient) SubmitBeaconBlock(block *spec.VersionedBeaconBlock, sig phas
 	return gc.client.SubmitBeaconBlock(gc.ctx, signedBlock)
 }
 
+func (gc *goClient) SubmitValidatorRegistration(pubkey []byte, feeRecipient bellatrix.ExecutionAddress, sig phase0.BLSSignature) error {
+	if gc.batchRegistration {
+		return gc.handleBatchValidatorRegistration(pubkey, feeRecipient, sig)
+	}
+
+	return gc.client.SubmitValidatorRegistrations(gc.ctx, []*api.VersionedSignedValidatorRegistration{
+		gc.createValidatorRegistration(pubkey, feeRecipient, sig),
+	})
+}
+
+func (gc *goClient) handleBatchValidatorRegistration(pubkey []byte, feeRecipient bellatrix.ExecutionAddress, sig phase0.BLSSignature) error {
+	gc.batchMu.Lock()
```

Comment:
should we lock all this function for sure? think if locking only when accessing a state that should be protected is possible.